addons/web/static/src/core/utils/ui.js

```diff
@@ -227,3 +227,349 @@ export function addLoadingEffect(btnEl) {
         loaderEl.remove();
     };
 }
+
+/**
+ * jQuery find function behavior is::
+ *
+ *      $('A').find('A B') <=> $('A A B')
+ *
+ * The searches behavior to find options' DOM needs to be::
+ *
+ *      $('A').find('A B') <=> $('A B')
+ *
+ * This is what this function does.
+ *
+ * @param {jQuery} $from - the jQuery element(s) from which to search
+ * @param {string} selector - the CSS selector to match
+ * @param {boolean} [addBack=false] - whether or not the $from element
+ *                                  should be considered in the results
+ * @returns {jQuery}
+ */
+
+// TODO: Need to convert cssFind from jquery to vanilla js
+
+export function cssFind($from, selector, addBack) {
+    var $results;
+
+    // No way to correctly parse a complex jQuery selector but having no
+    // spaces should be a good-enough condition to use a simple find
+    var multiParts = selector.indexOf(' ') >= 0;
+    if (multiParts) {
+        $results = $from.closest('body').find(selector).filter((i, $el) => $from.has($el).length);
+    } else {
+        $results = $from.find(selector);
+    }
+
+    if (addBack && $from.is(selector)) {
+        $results = $results.add($from);
+    }
+
+    return $results;
+}
+
+/**
+ * Renders a button with standard odoo template. This does not use any xml
+ * template to avoid forcing the frontend part to lazy load a xml file for
+ * each widget which might want to create a simple button.
+ *
+ * @param {Object} options
+ * @param {Object} [options.attrs] - Attributes to put on the button element
+ * @param {string} [options.attrs.type='button']
+ * @param {string} [options.attrs.class='btn-secondary']
+ *        Note: automatically completed with "btn btn-X"
+ *        (@see options.size for the value of X)
+ * @param {string} [options.size] - @see options.attrs.class
+ * @param {string} [options.icon]
+ *        The specific fa icon class (for example "fa-home") or an URL for
+ *        an image to use as icon.
+ * @param {string} [options.text] - the button's text
+ * @returns {HTMLElement}
+ */
+export function renderButton(options) {
```

Comment:
Do we still need this function? Maybe we could have a <Button> component instead